src/test/kotlin/org/koil/image/FileBackedImageResizerTest.kt

```diff
@@ -0,0 +1,32 @@
+package org.koil.image
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import org.junit.jupiter.api.Test
+
+class FileBackedImageResizerTest {
+    val resizer = FileBackedImageResizer()
+    val png = this::class.java.getResourceAsStream("/img/large.png")!!
+    val jpg = this::class.java.getResourceAsStream("/img/large.jpg")!!
+    val scaledPng = this::class.java.getResourceAsStream("/img/scaled-png.png")!!
+    val scaledJpg = this::class.java.getResourceAsStream("/img/scaled-jpg.png")!!
+
+    @Test
+    internal fun `convert jpg to png`() {
+        val result = resizer.resize(jpg)
```

Comment:
I think that the method should be renamed to `resizeAndConvertToPng`. It won't return the explicit image type because the output of the method is an InputStream.

Reply:
I've also created more meaningful parameters. While it might be a little overengineered for now, it's not much work that's made this more readable and flexible imo.